cpp/src/arrow/acero/meson.build

```
@@ -80,12 +80,12 @@ arrow_acero_srcs = [
 arrow_acero_lib = library(
     'arrow-acero',
     sources: arrow_acero_srcs,
-    dependencies: [arrow_dep],
+    dependencies: [arrow_dep, arrow_compute_dep],
```

`arrow_compute_dep` must be placed before `arrow_dep` because `arrow_compute_dep` depends on `arrow_dep`:

```suggestion
    dependencies: [arrow_compute_dep, arrow_dep],
```